### PR TITLE
Fix + add new view model tests

### PR DESCRIPTION
**Pull request by juaniac**
This PR is meant:
-  to fix old view model tests that don't work 100% of the time due to the database changes.
- to add new view model tests to add coverage.
- to fix CI that runs 2 times and modify the database simultaneously which caused errors.

### CODE REVIEW

**Comment by juaniac** [comment]
In FindEventViewModel:
image
This doesn't follow the coding guidelines for view models as the view model shouldn't care about what the UI screen does exactly. So loading should always be set to false inside the view model. 
I will create a separate variable for the UI recomposition in the FindEvents UI screen file.

**Review by leagrieder — COMMENTED** [review]
1. CI - Test Runner Workflow Adjustment
   -  Good job on adding the  '**' wildcard to the branches array. This change means that the CI tests will run on pushes to any branch, not just 'main'. This could potentially increase the CI server load but ensures early detection of issues across all branches.

2. The code is well-structured and follows Kotlin's conventions. Usage of constants and well-named variables enhances readability and maintainability.

3. The PR includes extensive adjustments to testing configurations, thanks for writing those!

4. This PR shows a promising direction in enhancing the ChimpagneApp’s functionality and user experience.

5. For future references, when you create draft pull requests, could you explain what it is that you are still working on (so we have an idea oh what still needs to be done) ? :)

File: app/src/androidTest/java/com/monkeyteam/chimpagne/newtests/TestUtils.kt [hunk 1]
```diff
@@ -20,36 +21,58 @@ val TEST_EVENTS =
             id = "FIRST_EVENT",
             title = "First event",
             description = "a random description",
-            location = Location("Paris"),
+            location = Location("EPFL", 46.519124, 6.567593),
             public = true,
-            tags = listOf("vegan"),
+            /* DO NOT DELETE OR MODIFY TAGS*/
```
Comment (Athanaze) [inline app/src/androidTest/java/com/monkeyteam/chimpagne/newtests/TestUtils.kt:26]: I appreciate the insightful comment and suggest that next time you use single line comments with // when the comment is, well, in a single line.

**Review by Athanaze — APPROVED** [review]
Overall, this PR is quite readable, and the code works.

The added comments are important to prevent issues down the line, and offer good guidance for the more advanced end-to-end tests that we will need to do.

With the PR, we are getting close to our test coverage goal :sunglasses:

**Pull request merged**